src/v/cluster/partition_balancer_backend.h

```diff
@@ -49,7 +49,11 @@ class partition_balancer_backend {
     bool is_leader() const { return _raft0->is_leader(); }
 
     std::optional<model::node_id> leader_id() const {
-        return _raft0->get_leader_id();
+        auto leader_id = _raft0->get_leader_id();
+        if (leader_id && leader_id == _raft0->self().id() && !is_leader()) {
+            return std::nullopt;
+        }
+        return leader_id;
```

Comment:
@ZeDRoman @ztlpn 

quick question: is this the same as the `is_elected_leader()` interface?

Reply:
You mean that `leader_id == _raft0->self().id() && !is_leader()` is true only if `is_elected_leader()`?

Reply:
Well this situation is equivalent to `is_elected_leader && !is_leader`.  In this case we want to return `nullopt` even if `get_leader_id` returns non-null value.

Reply:
ahh ok i see it now. i'm not sure where i was getting confused. thanks!